pkg/ipc/ipc.go

```diff
@@ -645,7 +645,14 @@ func makeCommand(pid int, bin []string, config *Config, inFile, outFile *os.File
 	if err := cmd.Start(); err != nil {
 		return nil, fmt.Errorf("failed to start executor binary: %v", err)
 	}
+	c.exited = make(chan error, 1)
 	c.cmd = cmd
+	go func(c *command) {
+		err := c.cmd.Wait()
+		c.exited <- err
+		close(c.exited)
+		rp.SetDeadline(time.Now())
+	}(c)
```

Comment:
Humm... File.Fd docs say:
```
On Unix systems this will cause the SetDeadline methods to stop working.
```
https://pkg.go.dev/os#File.Fd

and I see that os.startProcess calls Fd()
so does this work?

This code is super complex, I wonder if it's possible to unit test these scenarios. We could start executor with some special flags that will make it produce all kinds on bad things (fail handshake, leak descriptors, etc).

Reply:
os.startProcess calls Fd() on wp, but we are working with rp. I have added a test.

Reply:
Right! OK, let's see how it works. Thanks for the test.